### PR TITLE
Ability to handle multiple roles in workflow transitions

### PR DESCRIPTION
**Pull request by mittulmadaan**
- [ Enable multi-select of roles for workflow transitions.]

- [ Handled duplicates.]

- [ Synced changes with ReadOnlyArray changes. ]  

- [ Handled dashed border style for Workflow -> Roles. ]

### CODE REVIEW

File: tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs [hunk 1]
```diff
@@ -19,8 +24,33 @@ public void Should_serialize_and_deserialize()
             var workflow = Workflow.Default;
 
             var serialized = workflow.SerializeAndDeserialize();
+            var test = workflow.GetTransitions(Status.Archived);
```
Comment (SebastianStehle) [inline tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs]: Whats that used for?

File: tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs [hunk 2]
```diff
@@ -19,8 +24,33 @@ public void Should_serialize_and_deserialize()
...
+            };
+
+            var serializerSettings = new JsonSerializerSettings
+            {
+                ContractResolver = new ConverterContractResolver()
+            };
+
+            var json = JsonConvert.SerializeObject(source, serializerSettings);
+
+            var serialized = JsonConvert.DeserializeObject<JsonWorkflowTransition>(json);
```
Comment (SebastianStehle) [inline tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs]: Plese use the SerializeAndDeserialize extension method for or the Helper class in general and not JSON.NET directly

File: tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs [hunk 3]
```diff
@@ -19,8 +24,33 @@ public void Should_serialize_and_deserialize()
...
             var serialized = workflow.SerializeAndDeserialize();
+            var test = workflow.GetTransitions(Status.Archived);
 
             serialized.Should().BeEquivalentTo(workflow);
         }
+
+        [Fact]
+        public void Should_verify_roles_mapping_in_workflow_transition()
+        {
+            var source = new
```
Comment (SebastianStehle) [inline tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs]: I prefer to have this single lined when there are only a few properties.

File: src/Squidex/app/features/settings/pages/workflows/workflow-transition.component.html [hunk 4]
```diff
@@ -21,17 +21,21 @@
...
-            (ngModelChange)="changeRole($event)">
-            <option></option>
-            <option *ngFor="let role of roles; trackBy: trackByRole" [ngValue]="role.name">{{role.name}}</option>
-        </select>
-
-        <span class="select-placeholder" *ngIf="!transition.role || transition.role === ''">Add Role</span>
+            [dashed]="true"
+            [allowDuplicates]="false"
+            [suggestions]="roleSuggestions"
+            singleLine="true"
```
[the remark this answers is not in the document]
Comment (mittulmadaan) [reply src/Squidex/app/features/settings/pages/workflows/workflow-transition.component.html]: Updated.

File: src/Squidex/Areas/Api/Controllers/Apps/Models/WorkflowTransitionDto.cs [hunk 5]
```diff
@@ -19,7 +20,7 @@ public sealed class WorkflowTransitionDto
         /// <summary>
         /// The optional restricted role.
         /// </summary>
-        public string Role { get; set; }
+        public IReadOnlyList<string> Roles { get; set; }
```
[the remark this answers is not in the document]
Comment (mittulmadaan) [reply src/Squidex/Areas/Api/Controllers/Apps/Models/WorkflowTransitionDto.cs]: I've updated it. Thanks for the suggestion Sebastian.

File: tests/Squidex.Domain.Apps.Entities.Tests/Contents/DynamicContentWorkflowTests.cs [hunk 6]
```diff
@@ -44,7 +44,7 @@ public class DynamicContentWorkflowTests
                         new Dictionary<Status, WorkflowTransition>
                         {
                             [Status.Archived] = new WorkflowTransition(),
-                            [Status.Published] = new WorkflowTransition("data.field.iv === 2", "Editor")
+                            [Status.Published] = new WorkflowTransition("data.field.iv === 2", new[] { "Editor" })
```
[the remark this answers is not in the document]
Comment (mittulmadaan) [reply tests/Squidex.Domain.Apps.Entities.Tests/Contents/DynamicContentWorkflowTests.cs]: I've added a test case.

File: src/Squidex.Domain.Apps.Core.Model/Contents/Json/JsonWorkflowTransition.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    public class JsonWorkflowTransition
+    {
+        [JsonProperty]
+        public string Expression { get; set; }
+
+        [JsonProperty]
+        public string Role { get; set; }
+
+        [JsonProperty]
+        public IReadOnlyList<string> Roles { get; }
```
[the remark this answers is not in the document]
Comment (mittulmadaan) [reply src/Squidex.Domain.Apps.Core.Model/Contents/Json/JsonWorkflowTransition.cs]: I've updated it. Thanks for the suggestion Sebastian.

File: src/Squidex.Domain.Apps.Core.Model/Contents/WorkflowTransition.cs [hunk 8]
```diff
@@ -13,13 +15,13 @@ public sealed class WorkflowTransition
 
         public string Expression { get; }
 
-        public string Role { get; }
+        public IReadOnlyList<string> Roles { get; }
```
[the remark this answers is not in the document]
Comment (mittulmadaan) [reply src/Squidex.Domain.Apps.Core.Model/Contents/WorkflowTransition.cs]: I've updated it. Thanks for the suggestion Sebastian.

File: tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs (hunk 1 above)
Comment (mittulmadaan) [reply tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs]: Sorry, that was a typo from my side. Apologies.

File: tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs (hunk 2 above)
Comment (mittulmadaan) [reply tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs]: I've updated it, it's good to utilize extension methods throughout. Thanks.

File: tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs (hunk 3 above)
Comment (mittulmadaan) [reply tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/WorkflowJsonTests.cs]: I've updated it.

**Pull request merged**